### PR TITLE
environment variables cannot be picked up by the applications automatically

### PR DESCRIPTION
**Pull request by xnslx**

### CODE REVIEW

**Comment by xnslx** [comment]
The environment variable cannot be picked up by Remix.

**Comment by cjreysayago** [comment]
Hey @xnslx I hope you're doing great.
I was taking a look at the diff of this PR and trying to understand why is not taking the env vars to you. 

What's your current status? Are you trying to run this in local and got a similar exception as this one?
CleanShot 2023-10-03 at 16 26 42

If so, then all you have to do is rename your Sanity env vars using the "PUBLIC" prefix as mentioned here: 

CleanShot 2023-10-03 at 16 31 04

**Comment by bjohns1982** [comment]
TL;DR --> **The project is looking for the word PUBLIC in the constants in the `.env` file**
@cjreysayago's response helped a bit as I was getting the "Configuration must contain projectId" error also. The README indicates the following: 
image

It's possible that the process to populate the `.env` file is not writing the correct names for those constants 🤷 

I looked more closely at the stack trace and where the error was coming from and that led me to ```/app/sanity/client.ts``` which imports ```/app/sanity/projectDetails.ts```. Within `projectDetails` I noticed the names of the constants didn't match up with what was in my `.env` file.

**original broken `.env` file:**
```
SANITY_PROJECT_ID="14131gfw"
SANITY_DATASET="production"
```

**new working `.env` file:**
```
SANITY_PUBLIC_PROJECT_ID="12345"
SANITY_PUBLIC_DATASET="prod"
```

**Comment by xnslx** [comment]
> Hey @xnslx I hope you're doing great. I was taking a look at the diff of this PR and trying to understand why is not taking the env vars to you.
> 
> What's your current status? Are you trying to run the app in your local and got a similar exception as this one?
> 
> CleanShot 2023-10-03 at 16 26 42
> 
> If so, then you likely have faulty names for your env vars in your `.env` file. I will highly advise that you double-check your env file and ensure that the `PUBLIC` prefixed variables such as: `SANITY_PUBLIC_PROJECT_ID` and `SANITY_PUBLIC_DATASET` are indeed there along with their values.
> 
> CleanShot 2023-10-03 at 16 31 04

Yes! I think `PUBLIC` should be reason why env variables cannot be picked up automatically. You are very helpful.

**Comment by cjreysayago** [comment]
Glad to help 🙂

**Comment by SimeonGriggs** [comment]
Sorry about this these variables were misnamed compared to those generated by the init script. Will be updated.
